parts/22-query.sh

```diff
@@ -4905,3 +4905,67 @@ query_tpt-tool-memory() { ##? [--startyear=<YYYY>] [--endyear=<YYYY>] [--formula
 	EOF
 }
 
+query_tools-usage-per-month() { ##? [--startmonth=<YYYY>-<MM>] [--endmonth=<YYYY>-<MM>] [--toolnames=<tool1>,<tool2>,...]: toolnames is required. startmonth is 1 year ago and end month is current month by default
+	meta <<-EOF
+		AUTHORS: lldelisle
+		ADDED: 22
+	EOF
+	handle_help "$@" <<-EOF
+		Tools Usage Tracking: cpu-hours and nb_users by Month-Year.
+
+		    $ gxadmin query tools-usage-per-month --toolnames=bowtie2,cufflinks
+		       month    | cpu_hours | tool_name | nb_users 
+			------------+-----------+-----------+----------
+			 2023-12-01 |    135.08 | cufflinks |        3
+			 2023-11-01 |    105.13 | bowtie2   |        1
+			 2023-10-01 |   1085.34 | bowtie2   |        2
+			 2023-10-01 |    248.37 | cufflinks |        1
+			 2023-09-01 |     11.78 | bowtie2   |        1
+			 2023-09-01 |    233.03 | cufflinks |        1
+			 2023-08-01 |    796.15 | bowtie2   |        2
+			 2023-08-01 |     18.84 | cufflinks |        1
+			 2023-07-01 |     20.04 | bowtie2   |        1
+			 2023-06-01 |    271.16 | bowtie2   |        3
+			 2023-06-01 |    188.25 | cufflinks |        2
+			 2023-05-01 |    732.74 | bowtie2   |        3
+			 2023-05-01 |     23.14 | cufflinks |        1
+			 2023-04-01 |    426.32 | bowtie2   |        2
+			 2023-03-01 |    944.02 | bowtie2   |        2
+			 2023-02-01 |     52.95 | bowtie2   |        2
+			 2022-12-01 |    135.82 | bowtie2   |        1
+			 (17 rows)
+	EOF
+	
+	filter_by_time_period=""
+	if [[ -n $arg_startmonth ]]; then
+		filter_by_time_period="date_trunc('month', job.create_time AT TIME ZONE 'UTC') >= '$arg_startmonth-01'::date"
+	else
+		filter_by_time_period="job.create_time > NOW() - interval '1 year'"
+	fi
+	if [[ -n $arg_endmonth ]]; then
+		filter_by_time_period=$filter_by_time_period "AND date_trunc('month', job.create_time AT TIME ZONE 'UTC') <= '$arg_endmonth-01'::date"
```

Comment:
```suggestion
		filter_by_time_period="$filter_by_time_period AND date_trunc('month', job.create_time AT TIME ZONE 'UTC') <= '$arg_endmonth-01'::date"
```

otherwise it isn't set correctly